controllers/api.js

```diff
@@ -722,7 +724,20 @@ exports.getGoogleDrive = (req, res) => {
   drive.files.list({
     fields: 'files(iconLink, webViewLink, name)'
   }, (err, response) => {
-    if (err) return console.log(`The API returned an error: ${err}`);
+    if (err) {
+      if (err.code === 401) {
+        refresh.requestNewAccessToken('google', token.refreshToken, (err, newAccessToken) => {
+          if (err) return console.log(`Requesting new access token failed: ${err}`);
+          User.updateOne({
+            _id: req.user.id,
+            'tokens.accessToken': token.accessToken
+          }, {
+            $set: { 'tokens.$.accessToken': newAccessToken }
+          }).then(() => res.redirect('/api/google/drive')).catch(err => console.log(`User update returned an error: ${err}`));
+        });
+      }
+      return console.log(`The API returned an error: ${err}`);
```

Comment:
The `console.log(`The API returned an error: ${err}`);`  seems to run even if the issue was the need for a refresh token and if the few lines before it was able to successfully refresh the token and render the Google Drive files.